src/tiramisu_function.cpp

```diff
@@ -116,6 +116,112 @@ isl_union_map *tiramisu::function::compute_dep_graph() {
     return result;
 }
 
+const std::map<std::string, tiramisu::buffer *> tiramisu::function::get_mapping() const
+{
+  return this->mapping;
+
+}
+
+void  tiramisu::function::add_mapping(std::pair<std::string,tiramisu::buffer *> p)
+{
+  this->mapping.insert(p);
+
+}
+
+	
+const int  &function::Automatic_communication(tiramisu::computation* c1,tiramisu::computation* c2 ) const
```

Comment:
Can you document how this function works? The high level picture of the algorithm.